### DISCUSSION TITLE
Default to "confirm_os_window_close 1" rather than 0

### DISCUSSION THREAD

**Discussion by jab** [kovidgoyal/kitty#4658] [Ideas]
"1" would be a safer default, and would better match the behavior of other terminals that many users are migrating from, e.g. Terminal.app, iTerm, etc. Such users are likely to assume that Kitty does this by default too, and only find out the hard way that it doesn't, once they've already lost work. Case in point: this just happened to me, and it was not a user-friendly experience.

Would you accept a PR to make this change?

Thanks for your consideration and for the great work on Kitty.

------

Example of Terminal.app's default behavior when trying to close a window with an active process:

<img width="850" alt="Screen Shot 2022-02-08 at 5 46 52 PM" src="https://user-images.githubusercontent.com/64992/153088687-4b06f10f-0a07-48bf-ab94-00f83958500c.png">

Screenshot of Terminal.app's default config for this behavior:

<img width="779" alt="Screen Shot 2022-02-08 at 5 47 45 PM" src="https://user-images.githubusercontent.com/64992/153088725-165f46df-669f-467f-9b34-496866aac0c1.png">

**Comment by kovidgoyal** [comment]
No, I wouldn't. I dislike confirmation prompts that second guess my
intentions. Since I open and close *lots* of terminals (as do many
terminal power users) the cost fo havong to confirm every close is too
high.

**Comment by jab** [comment]
See the second screenshot in the OP, in particular the "Only if there are processes other than the login shell and..." option which is the default behavior in Terminal, as it should be in Kitty.

If Kitty's close confirmation does not yet know how to distinguish between "I am running vim and would lose work if this window is closed" and "the only process I'm running is my login shell", then that is worth fixing first. Until that's fixed, I agree that setting "confirm_os_window_close 1" would be the wrong default. I also am a terminal power user who opens lots of windows and it would drive me nuts to have to confirm each close. However, a close that would kill running *non-login-shell* processes should require confirmation by default. Of all terminals, the one named Kitty should most want to make "The cat walked over my keyboard!" more likely a non-destructive operation :)

**Comment by kovidgoyal** [comment]
confirm_os_window_close -1
